eth/gasestimator/gasestimator.go

```diff
@@ -94,13 +96,13 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 
 		// If the allowance is larger than maximum uint64, skip checking
 		if allowance.IsUint64() && hi > allowance.Uint64() {
-			transfer := call.Value
-			if transfer == nil {
-				transfer = new(big.Int)
-			}
```

Comment:
The previous code, at least, we instantiated `transfer` close to the one single point where we needed it. Your new code is possibly slightly more clever, in that it uses an exising if-clause, so we basically shave off a nil-comparison. However, that 'cleverness' means that the variable declaration/instantiatin becomes smeared over a larger scope.

Plus you make it a string, while the actual thing it represents is inherently numeric.